private/buf/cmd/buf/command/format/format.go

```diff
@@ -0,0 +1,388 @@
+	Write           bool
+	// special
+	InputHashtag string
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	bufcli.BindInputHashtag(flagSet, &f.InputHashtag)
+	bufcli.BindPaths(flagSet, &f.Paths, pathsFlagName)
+	bufcli.BindExcludePaths(flagSet, &f.ExcludePaths, excludePathsFlagName)
+	bufcli.BindDisableSymlinks(flagSet, &f.DisableSymlinks, disableSymlinksFlagName)
+	flagSet.BoolVarP(
+		&f.Diff,
+		diffFlagName,
+		diffFlagShortName,
+		false,
+		"Display diffs instead of rewriting files.",
+	)
+	flagSet.BoolVarP(
+		&f.Write,
+		writeFlagName,
+		writeFlagShortName,
+		false,
+		"Rewrite files in-place.",
+	)
+	flagSet.StringVar(
+		&f.ErrorFormat,
+		errorFormatFlagName,
+		"text",
+		fmt.Sprintf(
+			"The format for build errors printed to stderr. Must be one of %s.",
+			stringutil.SliceToString(bufanalysis.AllFormatStrings),
+		),
+	)
+	flagSet.StringVarP(
+		&f.Output,
+		outputFlagName,
+		outputFlagShortName,
+		"-",
+		fmt.Sprintf(
+			`The output location for the formatted files. Must be one of format %s. If omitted, the result is written to stdout.`,
+			buffetch.SourceFormatsString,
+		),
+	)
+	flagSet.StringVar(
+		&f.Config,
+		configFlagName,
+		"",
+		`The file or data to use for configuration.`,
+	)
+}
+
+func run(
+	ctx context.Context,
+	container appflag.Container,
+	flags *flags,
+) (retErr error) {
+	if err := bufcli.ValidateErrorFormatFlag(flags.ErrorFormat, errorFormatFlagName); err != nil {
+		return err
+	}
+	if flags.Output != "-" && flags.Write {
+		return fmt.Errorf("--%s cannot be used with --%s", outputFlagName, writeFlagName)
+	}
+	input, err := bufcli.GetInputValue(container, flags.InputHashtag, ".")
+	if err != nil {
+		return err
+	}
+	sourceOrModuleRef, err := buffetch.NewRefParser(container.Logger(), buffetch.RefParserWithProtoFileRefAllowed()).GetSourceOrModuleRef(ctx, input)
+	if err != nil {
+		return err
+	}
+	if _, ok := sourceOrModuleRef.(buffetch.ModuleRef); ok && flags.Write {
+		return fmt.Errorf("--%s cannot be used with module reference inputs", writeFlagName)
+	}
+	registryProvider, err := bufcli.NewRegistryProvider(ctx, container)
+	if err != nil {
+		return err
+	}
+	moduleReader, err := bufcli.NewModuleReaderAndCreateCacheDirs(container, registryProvider)
+	if err != nil {
+		return err
+	}
+	runner := command.NewRunner()
+	storageosProvider := bufcli.NewStorageosProvider(flags.DisableSymlinks)
+	moduleConfigReader, err := bufcli.NewWireModuleConfigReaderForModuleReader(
+		container,
+		storageosProvider,
+		runner,
+		registryProvider,
+		moduleReader,
+	)
+	if err != nil {
+		return err
+	}
+	moduleConfigs, err := moduleConfigReader.GetModuleConfigs(
+		ctx,
+		container,
+		sourceOrModuleRef,
+		flags.Config,
+		flags.Paths,
+		flags.ExcludePaths,
+		false,
+	)
+	if err != nil {
+		return err
+	}
+	var readWriteBucket storage.ReadWriteBucket
+	if flags.Output != "-" {
+		if err := os.MkdirAll(flags.Output, 0755); err != nil {
```

Comment:
Discussed offline - 

We eventually want to deprecate `bufwire` since it's kind of devolved into a catch-all place for functionality. Given that this functionality is specific to the format command, we'll leave this as-is.

The single file input issue was addressed.